src/System.Management.Automation/engine/ExecutionContext.cs

```diff
@@ -1269,55 +1269,116 @@ internal PSTransactionManager TransactionManager
 
         internal PSTransactionManager transactionManager;
 
-        internal Assembly AddAssembly(string name, string filename, out Exception error)
-        {
-            Assembly loadedAssembly = LoadAssembly(name, filename, out error);
-
-            if (loadedAssembly == null)
-                return null;
-
-            if (AssemblyCache.ContainsKey(loadedAssembly.FullName))
+        /// <summary>
+        /// This method is used for assembly loading requests stemmed from 'InitialSessionState' binding and module loading.
+        /// </summary>
+        /// <param name="source">Source of the assembly loading request, usually is a module name.</param>
+        /// <param name="assemblyName">Name of the assembly to be loaded.</param>
+        /// <param name="fileName">Path of the assembly to be loaded.</param>
+        /// <param name="error">Exception that is caught when the loading fails.</param>
+        internal Assembly AddAssembly(string source, string assemblyName, string fileName, out Exception error)
```

Comment:
`fileName` parameter should be `filePath` or `filePathName` to make it clear it is the full unique file name path, which is important for distinguishing assemblies in the cache.